### PR TITLE
Scaffold frontpage

### PR DESCRIPTION
**Pull request by mikeyrayvon**

### CODE REVIEW

File: front-page.php [hunk 1]
```diff
@@ -10,26 +12,66 @@
...
-if( have_posts() ) {
-  while( have_posts() ) {
-    the_post();
+  $args = array(
+    'post_type' => 'programacion',
+    'posts_per_page' => 1,
+    'meta_query' => array(
+      'relation' => 'AND',
+      array(
+        'key'     => '_igv_start_time',
```
Comment (varffs) [inline front-page.php]: shouldnt this be end time otherwise after the start of the latest it will not show?

File: front-page.php [hunk 2]
```diff
@@ -10,26 +12,66 @@
...
 
     <article <?php post_class(); ?> id="post-<?php the_ID(); ?>">
 
-      <a href="<?php the_permalink() ?>"><?php the_title(); ?></a>
+        <?php if (!empty($meta['_igv_start_time'][0])) { 
+            $start_time = $meta['_igv_start_time'][0];
+          ?>
+        <div>
+          <div><?php echo __('[:es]Desde[:en]From'); ?></div>
+          <div><?php echo date( 'M', $start_time ) . '. ' . date( 'd', $start_time ); ?></div>
```
Comment (varffs) [inline front-page.php]: if `$start_time` is not set this will error I think?

File: front-page.php [hunk 3]
```diff
@@ -10,26 +12,66 @@
...
 
-      <?php the_content(); ?>
+        <h3><?php if (!empty($meta['_igv_subtitle'][0])) { echo $meta['_igv_subtitle'][0];} ?></h3>
+
+        <?php if (!empty($meta['_igv_end_time'][0])) { 
+            $end_time = $meta['_igv_end_time'][0];
+          ?>
+        <div>
+          <div><?php echo __('[:es]Hasta[:en]Until'); ?></div>
+          <div><?php echo date( 'M', $end_time ) . '. ' . date( 'd', $end_time ); ?></div>
```
Comment (varffs) [inline front-page.php]: you can put the formatting in the data function like `('M. d')`

File: front-page.php (hunk 2 above)
Comment (mikeyrayvon) [inline front-page.php]: but its inside the `if`

File: front-page.php (hunk 1 above)
Comment (mikeyrayvon) [inline front-page.php]: i guess im misunderstanding the compare.  which side does the key go on?

Comment (varffs) [inline front-page.php]: `<=` less than or equal to I think

Comment (mikeyrayvon) [inline front-page.php]: ok the key goes on the left side of the comparison.  so `_igv_start_time <= $date` giving us the current, or most recent past entry.

Comment (varffs) [inline front-page.php]: ah rite yes im pretty sure thats how it goes

**Comment by varffs** [comment]
but now it only shows until the event starts again?

**Comment by mikeyrayvon** [comment]
until the event starts again? what?

**Comment by mikeyrayvon** [comment]
oh i also need to add something that lists the results by start date

**Comment by varffs** [comment]
The query says get posts with a start time less than or equal to now. So if the event start time is in the past [but end time is in the future] it is not found by this query.

I think you need to move the query out of the meta query array as the AND https://github.com/interglobalvision/estanciafemsa-mx/blob/scaffold-frontpage/front-page.php#L19 probably is stopping it from working.

**Comment by mikeyrayvon** [comment]
are you testing this locally?

**Comment by varffs** [comment]
No just reading the code and I guess I'm assuming it is causing you issues. I'll test it inna sec

**Comment by mikeyrayvon** [comment]
the code is working fine.  i think its correct as it.  test it

**Comment by mikeyrayvon** [comment]
try making an entry that starts before today and ends after today and publish that, then make another one that starts and ends before today and publish that.  then load the front page.  then delete the first one you made (the current event), and refresh the front page

**Comment by mikeyrayvon** [comment]
however i was just thinking.  its probably better to display Current, if no current Future, if no future Past

**Comment by varffs** [comment]
I think the query should look like:

```
$args = array(
    'post_type' => 'programacion',
    'posts_per_page' => 1,
    'meta_key' => '_igv_start_time',
    'orderby' => 'meta_value_num',
    'order' => 'DESC',
    'meta_compare' => '<=',
    'meta_value' => $date,
  );
```

And that works the same way. But neither make sense to me as I think the start time thing is still wrong. @mixcas should look with fresh eyes

**Comment by mixcas** [comment]
I think @mikeyrayvon queries are alright. Let me test.

**Pull request merged**